### ISSUE TITLE
Use clippy twice does not produce any warning

### ISSUE THREAD

**Issue by CyrilCalmels** [rust-lang-nursery/rust-clippy#2762]
Hello,
I have the same problem as https://github.com/rust-lang-nursery/rust-clippy/issues/2498 even if I use a version where it is supposed to be fixed:
```
> cargo clean
> cargo clippy -V
0.0.200
> cargo +nightly clippy
    Checking variables v0.1.0 (file:///----)
warning: equality checks against true are unnecessary
 --> src/main.rs:5:6
  |
5 |   if test == true { println!("true") }
  |      ^^^^^^^^^^^^ help: try simplifying it as shown: `test`
  |
  = note: #[warn(bool_comparison)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.200/index.html#bool_comparison

    Finished dev [unoptimized + debuginfo] target(s) in 0.23s
> cargo +nightly clippy
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
```

Thanks for any help.

**Comment by phansch** [comment]
Thanks for the report! This is currently tracked here: https://github.com/rust-lang-nursery/rust-clippy/issues/2604, so I'm going to close this as a duplicate.

You should be able to run `touch main.rs; cargo clippy` as a workaround.
